rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -96,6 +89,12 @@ if(NOT _generated_srv_py_files STREQUAL "")
   )
 endif()
 
+if(NOT _generated_msg_c_files STREQUAL "" OR NOT _generated_srv_c_files STREQUAL "")
+    foreach(_typesupport_impl ${_typesupport_impls})
+      list_append_unique(_generated_extension_${_typesupport_impl}_files "${_output_path}/_${PROJECT_NAME}_s.ep.${_typesupport_impl}.c")
+      list_append_unique(_generated_extension_files "${_generated_extension_${_typesupport_impl}_files}")
```

Comment:
While using `list_append_unique` works for both lines I would suggest using `list(APPEND` instead since there are by design no duplicates ever.